packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -1545,36 +1533,26 @@ function startTransition(setPending, config, callback) {
   }
 }
 
-function mountTransition(
-  config: SuspenseConfig | void | null,
-): [(() => void) => void, boolean] {
+function mountTransition(): [(() => void) => void, boolean] {
   const [isPending, setPending] = mountState(false);
-  const start = mountCallback(startTransition.bind(null, setPending, config), [
-    setPending,
-    config,
-  ]);
+  // The `start` method can be stored on a ref, since `setPending`
+  // never changes.
+  const start = startTransition.bind(null, setPending);
+  mountRef(start);
```

Comment:
This reminds me, should we update the hooks lint to not require passing startTransition to the dependencies?

Reply:
https://github.com/facebook/react/pull/19720